### ISSUE TITLE
Improve codegen time - compare grammar mtime to generated files

### ISSUE THREAD

**Issue by michaelneu** [tunnelvisionlabs/antlr4cs#245]
First of all: thank you for this runtime/codegen, it works like a charm (even on .net core).

My project contains four grammars (2x split-up lexer/parser grammars), and each time I hit *Build*, it takes quite some time to generate the lexers/parsers. I observed the build process and noticed that  the lexers/parsers get recreated on every build, even if their respective grammars didn't change. My observations didn't include going through the generator's source, but I suppose that there might not be a check whether re-generating is actually necessary. 

I was wondering, if you could compare the grammar's mtime to the generated lexer/parser mtime, and only recreate the lexer/parser when it's really necessary. 

In my current setup, I'm not using the plugin (on a previous project, I was only able to build when the plugin was installed, so this time I tried to not use it). When I change a grammar file, I need to rebuild the project for the changes to take effect. As *Rebuild* doesn't *Clean*, the previous files would still be there, and thus checking the mtime would also improve the build time there.

If you want, I can take a look into this issue and submit a PR (but probably not until next week).

**Comment by sharwell** [comment]
This is supposed to be handled by the msbuild tasks. I'll take a look near the end of the month and see if I can reproduce this.

**Comment by sharwell** [comment]
Duplicate of #289

**Comment by sharwell** [comment]
(I added a description of the problem and fix in #289 before realizing this earlier issue was the same thing.)
